test/org/jetbrains/java/decompiler/SingleClassesTest.java

```diff
@@ -211,6 +211,7 @@ protected void registerAll() {
     register(JAVA_8, "TestNativeMethods");
     register(JAVA_8, "TestThrowLoop");
     register(JAVA_8, "TestShiftLoop");
+    register(JASM, "TestDoubleCast");
```

Comment:
Can you add a `// TODO` comment describing the problem here and update the results file so the test passes? That way we keep the tests passing as regression tests and still document the known issues.

Reply:
Done